### DISCUSSION TITLE
Error while creating a PSTH

### DISCUSSION THREAD

**Discussion by CloeLhrx** [petersenpeter/CellExplorer#45] [Q&A]
Hello @petersenpeter,
I use this opportunity to thank you very much for putting together this awesome tool that makes complex data analysis very handy.
I am using CellExplorer to analyse Kilosort sorted data, manually curated with Phy. I manage to navigate in CellExplorer but  when it comes to compute PSTHs (spike times with optogenetic stimulation times for reference) the graph becomes empty with "no data" written (but no error messsage appears on matlab).

I have been trying to import a basename.eventname.events.mat file that corresponds to what CellExplorer wants, but I think I have some trouble finding the appropriate format.
Any help would be greatly appreciated.
[example.stim.events.zip](https://github.com/petersenpeter/CellExplorer/files/7675937/example.stim.events.zip)

**Comment by petersenpeter** [comment]
Hi

Your event file looks good to me. 

The event PSTHs are stored in the field `cell_metrics.events.eventName`, (e.g. cell_metrics.events.stim). So if that field is generated, it means that ProcessCellMetrics detected the events file and generated the PSTHs. Check the printed log when you run ProcessCellMetrics, that should give you an idea about this.

The PSTHs are generated with the script `calc_PSTH`, you could try to generate them directly. Load the events and spikes into Matlab and type:

```m
PSTH = calc_PSTH(stim,spikes)
```

If none of this works, please send me your spikes struct and the session struct as well for me to troubleshoot. 

/ Peter

**Reply by CloeLhrx** [reply]
Hello, 
Thanks for your fast response !

Yesterday, I was getting the field cell_metrics.events.stim on one computer but not another one so I deinstalled and reinstalled CellExplorer today and tried this command line 

>> cell_metrics = ProcessCellMetrics('session', session,'showGUI',true);  

but I now have this error and the field is not generated anymore:

"Error using vertcat
The following error occurred converting from double to struct: Conversion to struct from double is not possible.
Error in calc_PSTH (line 93)
[spike_times,index] = sort([spikes(j);event_times(:)]);
Error in ProcessCellMetrics (line 1059)
PSTH = calc_PSTH(eventOut.(eventName), spikes{spkExclu},'alignment', preferences.psth.alignment, 'smoothing',preferences.psth.smoothing, 'eventName', eventName,'binCount',preferences.psth.binCount, 'binDistribution', preferences.psth.binDistribution, 'duration',preferences.psth.duration, 'percentile', preferences.psth.percentile, 'plots',parameters.showFigures); "
            
If it can help, here is the .mat file with the session and cell_metrics struct
Best
Cloé
[session_and_spikes_test.zip](https://github.com/petersenpeter/CellExplorer/files/7687119/session_and_spikes_test.zip)

**Reply by petersenpeter** [reply]
I see an error on line 93, where the struct spikes are not indexed correctly. Are you using the latest version of CellExplorer? 
Your line: 
[spike_times,index] = sort([spikes(j);event_times(:)]);
Correct line: 
[spike_times,index] = sort([spikes.times{j};event_times(:)]);

Let me know if this solves your issue!

**Reply by CloeLhrx** [reply]
Thanks for your answer, I reinstalled it yesterday so I have the latest version now ! By changing the parentheses I get this :

Brace indexing is not supported for variables of this type.
Error in calc_PSTH (line 93)
    [spike_times,index] = sort([spikes{j};event_times(:)]);

**Reply by petersenpeter** [reply]
This is tricky for me to troubleshoot remotely, but it looks like your spike struct could be empty. Could you try running ProcessCellMetrics in the Matlab debug mode (The "Pause on Errors" mode). Then when the error occurs, check the content of the spike struct and the event_times. Does this happen with a particular unit (the j-index) or with any of them? 

If you want to we could do a quick zoom troubleshoot next week, e.g. Tuesday? If so send me an email with the suggested time and date (petersen.peter@gmail.com)

**Comment by petersenpeter** [comment]
I created an example dataset that might be helpful as well. You can find an events file there (Peter_MS22_180629_110319_concat.ripples.events.mat): 
https://buzsakilab.nyumc.org/datasets/PetersenP/CellExplorerExampleData/MS22/Peter_MS22_180629_110319_concat/
